prody2/viewers/viewer_gnm.py

```diff
@@ -0,0 +1,266 @@
+            title = 'Cross Correlation matrix'
+
+        plotter = EmPlotter(mainTitle=title)
+        plot = prody.showMatrix(matrix, vmax=1, vmin=-1, origin='lower')
+        
+        return [plotter] 
+
+    def _viewParam(self, paramName):
+        if paramName == 'displayModes':
+            # The following two lines display modes.sqlite file
+            # modes =  self.protocol.outputModes
+            # return [ObjectView(self._project, modes.strId(), modes.getFileName())]
+            # The following two lines display modes.xmd file
+            if isinstance(self.protocol, SetOfNormalModes):
+                modes = os.path.dirname(self.protocol[1].getModeFile()) + ".xmd"
+            else:
+                modes =  self.protocol._getPath("modes.xmd")
+            return [DataView(modes)]
+
+        elif paramName == 'displayMaxDistanceProfile':
+            if isinstance(self.protocol, SetOfNormalModes):
+                fn = os.path.dirname(os.path.dirname(self.protocol[1].getModeFile())) + "/extra/maxAtomShifts.xmd"
+            else:
+                fn = self.protocol._getExtraPath("maxAtomShifts.xmd")
+            return [createShiftPlot(fn, "Maximum atom shifts", "maximum shift")]
+
+    def _viewSQF(self, paramName):
+        """ visualization of square fluctuation for all/single mode/s. """   
+        modes_path = self.protocol._getPath("modes.gnm.npz")
+        modes = prody.loadModel(modes_path)
+        
+        cumulSqFluct = self.cumulSqFluct.get()
+        modeNumber = self.modeNumber.get()
+
+        if paramName == 'displaySqFlucts':
+            plotter = EmPlotter()
+            plot = prody.showSqFlucts(modes)
+        
+        elif paramName == 'displaySingleSqFluct':
+            if cumulSqFluct:
+                plotter = EmPlotter(mainTitle="Qumulative")
+                plot = prody.showSqFlucts(modes[:modeNumber])
+            else:
+                plotter = EmPlotter()
+                plot = prody.showSqFlucts(modes[modeNumber-1])
+
+        return [plotter]
+
+    
+    def _viewSingleMode(self, paramName):
+        """ visualization for a selected mode. """
+        if isinstance(self.protocol, SetOfNormalModes):
+            modes = self.protocol
+        else:
+            modes =  self.protocol.outputModes
+
+        modeNumber = self.modeNumber.get()
+        mode = modes[modeNumber]
+        
+        if mode is None:
+            return [self.errorMessage("Invalid mode number *%d*\n"
+                                      "Display the output Normal Modes to see "
+                                      "the availables ones." % modeNumber,
+                                      title="Invalid input")]
+                                      
+        elif paramName == 'displayVmd':
+            return [createVmdView(self.protocol, modeNumber)]
+        elif paramName == 'displayVmd2':
+            return [createVmdNmwizView(self.protocol, modeNumber)]
```

Comment:
This should be outside the single mode section and not have a modeNumber. NMWiz shows all modes